### ISSUE TITLE
Include module in the Companion core

### ISSUE THREAD

**Issue by hammady** [featherbear/bitfocus-companion-module-presonus-studiolive-iii#2]
I would like to use this module in the Companion core without having to manually install it. Are there plans to contact the guys in slack to create an official repo for the module?
I can help in the process because I have just authored a module and made it through the core.

**Comment by featherbear** [comment]
Hi @hammady, I won't be officially merging it into the companion monorepo until I get a stable implementation working - see the issue at https://github.com/featherbear/presonus-studiolive-api-demo/issues/1#issuecomment-1048412307

**Comment by hammady** [comment]
Thanks for the update. Does that mean metering is broken if I use the mentioned commit or later?
My planned usage is just to show the mute status for each channel and toggle them. Does this require metering?

**Comment by featherbear** [comment]
The implementation to get the _initial_ mute status (during initial connection) is broken, though it _can_ store *updates* to the mute states (i.e. pressing mute/unmute after being connected); 

Metering functionality is always optional to the functionality of the API.

I might attempt to push up a fix to allow the API to connect without waiting for initial state this weekend

**Comment by hammady** [comment]
Thanks again!

**Comment by featherbear** [comment]
@hammady Stable support for mute status and toggling has been implemented, for both the main, fx and aux mixes

bd7ac43310d8653558432463895ce29a23b2af55

**Comment by hammady** [comment]
Awesome. Thanks for the heads up!

**Comment by josephdadams** [comment]
@featherbear can we include this in the official betas? I can create the repo for you and get it merged

**Comment by featherbear** [comment]
> @featherbear can we include this in the official betas? I can create the repo for you and get it merged

@josephdadams I think the project is still in its alpha stage - there are some breaking changes yet to be made; and I'm also aware of some bugs, but I have minimal time outside of work to fix things (and provide technical support to Companion users). 

If you think that's fine, then I'm happy for you to create the repo to get the ball rolling 🥎

**Comment by josephdadams** [comment]
Do you know if scene recall is possible?

**Comment by featherbear** [comment]
Possible but currently unimplemented

**Comment by featherbear** [comment]
Project and scene recall has been implemented, but I'd like to get some testing feedback.

@josephdadams What's the best way to distribute an unofficial build of Companion with this module? I've been contacted by a few people via email RE this, who aren't confident in setting up a dev environment

**Comment by josephdadams** [comment]
From your dev environment, `yarn dist` (for all),  `yarn macdist` or `yarn windist` and distribute the binary files to them.

**Comment by featherbear** [comment]
@josephdadams 

Looking at the contents of the Electron build output, my module's dependencies aren't in the `node_modules` folder. Structure wise I have my module in the `module-local-dev` directory, with its dependencies in `module-local-dev/[...]/node_modules`. Do I need to touch anything in Companion's `package.json` so that the module dependencies are included in the build?

![image](https://user-images.githubusercontent.com/1159091/203673601-a56c653b-d3cd-4b80-b31a-b10a33bce27c.png)

**Comment by josephdadams** [comment]
Yes just npm install the dependencies to companion root package json. Just for dist purposes

**Comment by josephdadams** [comment]
It’s a known bug

**Comment by featherbear** [comment]
I probably won't be including this module in the Companion list because I don't have capacity to maintain this project past any "beta" version. Closing this issue out

I've got a release out that can be loaded in Companion 4
https://github.com/featherbear/bitfocus-companion-module-presonus-studiolive/releases/tag/v1.4.0
